libs/libc/stream/lib_stdsostream.c

```diff
@@ -63,6 +63,43 @@ static void stdsostream_putc(FAR struct lib_sostream_s *this, int ch)
   while (get_errno() == EINTR);
 }
 
+/****************************************************************************
+ * Name: stdsostream_puts
+ ****************************************************************************/
+
+static int stdsostream_puts(FAR struct lib_sostream_s *this,
+                            FAR const void *buffer, int len)
+{
+  FAR struct lib_stdsostream_s *sthis = (FAR struct lib_stdsostream_s *)this;
+  int result;
+  int errcode;
+
+  DEBUGASSERT(this && sthis->stream);
+
+  /* Loop until the character is successfully transferred or an irrecoverable
+   * error occurs.
+   */
+
+  do
+    {
+      result = lib_fwrite(buffer, len, sthis->stream);
+      if (result >= 0)
+        {
+          this->nput += result;
+          return result;
+        }
+
+      errcode = _NX_GETERRNO(result);
```

Comment:
result = _NX_GETERRNO(result);
and remove errcode

Reply:
done